agent/engine/docker_task_engine.go

```diff
@@ -723,14 +724,9 @@ func (engine *DockerTaskEngine) createContainer(task *api.Task, container *api.C
 	config.Labels[labelPrefix+"cluster"] = engine.cfg.Cluster
 
 	if dockerContainerName == "" {
-		name := ""
-		for i := 0; i < len(container.Name); i++ {
-			c := container.Name[i]
-			if !((c <= '9' && c >= '0') || (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c == '-')) {
-				continue
-			}
-			name += string(c)
-		}
+		// only alphanumeric and hyphen characters are allowed
+		reInvalidChars := regexp.MustCompile("[^A-Za-z0-9-]+")
```

Comment:
name is a bit confusing. Can just be `invalidChars`?

Reply:
re stands for regular expression, seems to be naming pattern for other regexp variables we have in the agent so I followed it.